esi_leap/api/controllers/v1/console_auth_token.py

```diff
@@ -0,0 +1,85 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import http.client as http_client
+import pecan
+from pecan import rest
+import wsme
+from wsme import types as wtypes
+import wsmeext.pecan as wsme_pecan
+
+from esi_leap.api.controllers import base
+from esi_leap.common import exception
+from esi_leap.common import ironic
+import esi_leap.conf
+from esi_leap.objects import console_auth_token as cat_obj
+
+CONF = esi_leap.conf.CONF
+
+
+class ConsoleAuthToken(base.ESILEAPBase):
+    node_uuid = wsme.wsattr(wtypes.text, readonly=True)
+    token = wsme.wsattr(wtypes.text, readonly=True)
+    access_url = wsme.wsattr(wtypes.text, readonly=True)
+
+    def __init__(self, **kwargs):
+        self.fields = ("node_uuid", "token", "access_url")
+        for field in self.fields:
+            setattr(self, field, kwargs.get(field, wtypes.Unset))
+
+
+class ConsoleAuthTokensController(rest.RestController):
+    @wsme_pecan.wsexpose(
+        ConsoleAuthToken, body={str: wtypes.text}, status_code=http_client.CREATED
+    )
+    def post(self, new_console_auth_token):
+        context = pecan.request.context
+        node_uuid_or_name = new_console_auth_token["node_uuid_or_name"]
+
+        # enable Ironic console
```

Comment:
Should enabling the console be a separate step? I think the process should be:

```
openstack baremetal node console enable <node>
openstack esi mumble <node>
```

(where `mumble` is whatever command is necessary to generate the token)

In other words, we shouldn't be tying the console enable state to whether or not there currently exists a valid token.

Reply:
I don't disagree with this, except I do think it's convenient to the user to simply be able to run a single command and get the token and the enabled console in one swoop. I think you mentioned in the demo meeting that the CLI command should perhaps be `openstack esi console enable` instead of `openstack esi console token create`, and I kind of like that.

I could put the console enablement in the CLI instead - well, more likely the SDK? Or I could rename this controller to `console.py` instead?

Reply:
Ah, that's definitely a fair point. I'll disentangle the two then.